06_puppy/nicholascross/main.go

```diff
@@ -0,0 +1,87 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+)
+
+type Puppy struct {
+	ID     uint64
+	Breed  string
+	Colour string
+	Value  float64
+}
+
+type Storer interface {
+	CreatePuppy(p Puppy)
+	RetrievePuppy(uint64) *Puppy
+	UpdatePuppy(uint64, Puppy)
+	DeletePuppy(uint64)
+}
+
+type MapStore map[uint64]Puppy
+
+var out io.Writer = os.Stdout
+
+func main() {
+	p := Puppy{1, "Jack Russel Terrier", "white and brown", 550.0}
+	fmt.Fprintf(out, "%d - %s [%s]: %f", p.ID, p.Breed, p.Colour, p.Value)
+}
+
+func (ms MapStore) CreatePuppy(p Puppy) {
+	if _, ok := ms[p.ID]; !ok {
+		ms[p.ID] = p
```

Comment:
In this context, the name of `ok` feels a bit strange. For Retrieve/Update/Delete it makes sense, for this, perhaps name it as `existing` or something like that.

Reply:
I have updated the variable name for all relevant cases from `ok` to `exists`

Reply:
Actually @alfredxiao , `ok` is the most common name used here, and referred to as the "comma ok" idiom (as described in Effective Go, in the Maps section: https://golang.org/doc/effective_go.html#maps). You'll see it used quite widely across the Go standard library.

However, I find `exists` to be ok to, so don't take this comment as a reason to change it back.